File: Core/src/main/kotlin/jokatu/components/stores/MessageStorer.java
package jokatu.components.stores;

import org.jetbrains.annotations.NotNull;
import org.springframework.messaging.Message;

/**
 * A place where messages can be stored.
 */
public interface MessageStorer {

	void store(@NotNull String destination, @NotNull Message message);

	void storeForUser(@NotNull String user, @NotNull String destination, @NotNull Message message);
}

File: Core/src/main/kotlin/jokatu/components/stores/MessageStorer.kt
package jokatu.components.stores

import jokatu.components.stomp.Destination
import org.springframework.messaging.Message

/**
 * A place where messages can be stored.
 */
interface MessageStorer {

	fun <T: Any> store(destination: Destination<T>, message: Message<T>)

	fun <T: Any> storeForUser(user: String, destination: Destination<T>, message: Message<T>)
}
